sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala

```diff
@@ -39,7 +39,7 @@ class JdbcRelationProvider extends CreatableRelationProvider
       null
     } else {
       JDBCPartitioningInfo(
-        partitionColumn, lowerBound.toLong, upperBound.toLong, numPartitions.toInt)
+        partitionColumn, lowerBound.toLong, upperBound.toLong, numPartitions.get)
```

Comment:
Thank you for review, @rxin . Yes. It implicitly depends on the partitionColumn requires all others columns.
I'll change like this. Is it better?
```
-    val partitionInfo = if (partitionColumn == null) {
+    val partitionInfo = if (partitionColumn == null || jdbcOptions == null || upperBound == null ||
+        numPartitions.isEmpty) {
```